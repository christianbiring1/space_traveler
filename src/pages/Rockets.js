import React from 'react';
import { useSelector } from 'react-redux';
import RocketContainer from '../components/RocketContainer';

function Rockets() {
  const rockets = useSelector((state) => state.rockets);

  return (
    <div data-testid="rocketsDisplay">
      {rockets.map((rocket) => (
        <RocketContainer
          key={rocket.id}
          id={rocket.id}
          name={rocket.rocket_name}
          description={rocket.description}
          image={rocket.flickr_image}
          reserved={rocket.reserved}
        />
      ))}
    </div>
  );
}

export default Rockets;
