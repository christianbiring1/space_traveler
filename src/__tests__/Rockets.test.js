import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Rockets from '../pages/Rockets';

describe('Rocket page tests', () => {
  it('Page renders as expected', () => {
    const component = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets />
        </BrowserRouter>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Page is visible to the user', () => {
    const { getByTestId } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Rockets />
        </BrowserRouter>
      </Provider>,
    );
    expect(getByTestId('rocketsDisplay')).toBeInTheDocument();
  });
});
