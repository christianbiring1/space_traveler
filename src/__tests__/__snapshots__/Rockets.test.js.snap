// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket page tests Page renders as expected 1`] = `
<div
  data-testid="rocketsDisplay"
/>
`;
